galpy/orbit/orbit_c_ext/integratePlanarOrbit.c

```diff
@@ -611,3 +618,35 @@ void evalPlanarRectDeriv_dxdv(double t, double *q, double *a,
   *a++= dFxdx * *(q+4) + dFxdy * *(q+5);
   *a= dFydx * *(q+4) + dFydy * *(q+5);
 }
+
+void initPlanarSplines(struct potentialArg * potentialArgs, double ** pot_args){
+
+  gsl_interp_accel *x_accel_ptr = gsl_interp_accel_alloc();
+  gsl_interp_accel *y_accel_ptr = gsl_interp_accel_alloc();
+  int nPts = (int) *(*pot_args+3);
+
+  gsl_spline *x_spline = gsl_spline_alloc(gsl_interp_cspline, nPts);
+  gsl_spline *y_spline = gsl_spline_alloc(gsl_interp_cspline, nPts);
+
+  double * t_arr = *pot_args+5;
+  double * x_arr = t_arr+1*nPts;
+  double * y_arr = t_arr+2*nPts;
+
+  double * t= (double *) malloc ( nPts * sizeof (double) );
+  double tf = *(*pot_args+2);
+  double to = *(*pot_args+1);
+
+  int i;
+  for (i=0; i<nPts; i++) *(t+i) = (t_arr[i]-to)/(tf-to);
+
+  gsl_spline_init(x_spline, t, x_arr, nPts);
+  gsl_spline_init(y_spline, t, y_arr, nPts);
+
+  potentialArgs->xSpline = x_spline;
+  potentialArgs->accx = x_accel_ptr;
+  potentialArgs->ySpline = y_spline;
+  potentialArgs->accy = y_accel_ptr;
+
+  potentialArgs->nargs= (int) 5+(1+*(*pot_args+4))*nPts;
```

Comment:
@turnergarrow: from looking at the code, it seems like after you've set up the splines in C, the only arguments that you need are ``amp``, ``t0``, and ``tf`` (none of the other arguments seem to be used in the ``MovingObjectPotential.c`` code, which would make sense). But currently, all of the arguments get copied into the ``potentialArgs-->args`` array. So the memory footprint is twice as large as necessary.

This is similar to what happens in ``interpRZPotential``. There, we deal with this by putting the arguments that need to be stored in ``potentialArgs-->args`` as the last ones in the argument array. Then you could increment the ``pot_args`` pointer until you get to ``amp`` (I think this is 
```
*(*pot_args)+=  (int) 2+(1+*(*pot_args+4))*nPts
```
 in this case, but it's a little complicated because ``pot_args`` is a double pointer, so maybe the syntax is wrong [haven't tested]) and then do
```
potentialArgs->nargs= 3;
```
because then only the last three arguments will get copied by the general code for copying ``pot_args`` to ``potentialArgs->args``.